Add mapcheck.py script

tnias opened the pull request:
This allows to check if hostname mappings in provided map.yml matches
the domain's `_onion-mx._tcp.domain.tld.` SRV records. (see SRV.md for
details)

tnias committed:
Add mapcheck.py script

This allows to check if hostname mappings in provided map.yml matches
the domain's `_onion-mx._tcp.domain.tld.` SRV records. (see SRV.md for
details)

ioparaskev commented:
Some thoughts:
- We haven't stated somewhere that entries in the mappings file should follow the SRV rule so maybe admins of those hostnames simply rely that since their entries are pinned, no other action is need from their site
- This script is a useful health check not only for admins relying on the mappings but for the project itself as well. Maybe we should add a weekly/daily travis cron job that will run this script to keep the mappings up to date

ioparaskev commented on scripts/mapcheck.py:
```diff
@@ -0,0 +1,51 @@
+            fails += 1
+
+# report back to the user
+if total == fails:
```
I would suggest that we print a warning everytime an entry does not match the mappings and keep quiet (or print a success message) when everything works as expected

ioparaskev commented on scripts/mapcheck.py:
```diff
@@ -0,0 +1,51 @@
+        try:
+            response = resolver.query(name, "SRV")
+            result = [str(x).rstrip(".").split(" ")[-1] for x in response]
+            print
```
I don't think we need this empty line print here

ioparaskev commented on scripts/mapcheck.py:
```diff
@@ -0,0 +1,51 @@
+else:
+    print("INFO {} of {} queries failed".format(fails, total), file=sys.stderr)
+
+exit(min(fails, 255))
```
not sure if it serves any purpose having the exit code == number-of-failures. Plus this works for only 255 entries (dream bigger :stuck_out_tongue: ) and if you have 1 failure, you get the same code as with invalid usage.. 3 exit codes (0=success, 1=invalid usage, 2=(fails !=0) are enough imho

ioparaskev commented on scripts/mapcheck.py:
```diff
@@ -0,0 +1,51 @@
+            result = [str(x).rstrip(".").split(" ")[-1] for x in response]
+            print
+            if onion in result:
+                print("OKAY {} -> {}".format(domain, onion))
```
Instead of printing every time something is okay/not-okay, maybe we could simply put the failed entries in a list and print those (and a message like `3/15 failed`) in the end. Main reason for this comment is to reduce noise (prints for no issue) and because I don't like `fails += 1` (reminds me of C++ :stuck_out_tongue_closed_eyes: )

tnias replied:
What shall I do about the `fails += 1`?

tnias committed:
mapcheck: reduce verbosity

ioparaskev commented on scripts/mapcheck.py:
```diff
@@ -0,0 +1,42 @@
+            fails += 1
+
+# report back to the user
+print("INFO {} of {} queries failed".format(fails, total), file=sys.stderr)
```
Instead of printing for each failure, you can use a list of failed and for each failure, append in the list the domain+onion (something like `fails.append("{} != {}".format(domain, onion))`. Then here in the end, 
```
if fails:
    print("INFO {} of {} queries failed".format(len(fails), total), file=sys.stderr)
    print("\n".join(b))
    exit(2)

exit(0)
```
So if everything is fine, no prints and exit(0), else print a statistic (x of y queries failed) and which failures did we have

tnias replied:
I don't see an advantage in putting them in a list first. But I will use exit(2) if there were any failures.

tnias committed:
mapcheck: change return code and clarify messages

Previously the messages commenting on failures were not too
userfriendly. They should sufficiently explain the issues.

Also we will now keep quiet if everything went smooth.
